consumer_monitor.go

```diff
@@ -265,16 +265,21 @@ func (m *consumerMonitor) updateConsumerStatus(fetchedOffsets map[string][]offse
 		offsets, ok := fetchedOffsets[topic.Name]
 		if !ok {
 			// No active subscriptions for the given topic at the time.
-			topic.partitionLag = map[int32]int64{}
+			topic.partitionLag = map[int32]partitionLagState{}
 			continue
 		}
 
 		for _, offset := range offsets {
 			// Only store the lag if it exceeds the configured threshold or is invalid.
 			// The next offset in the topic will always be greater than or equal to the last committed offset of the consumer.
+			// uncommittedOffsets flag is set to true in case the consumer group hasn't committed any offsets yet (returned value is -1) to prevent reporting wrong values to the client
```

Comment:
Let's extract a constant for the `-1` value. This will improve the readability of the code and we can also simplify the documentation.